rspace/src/main/scala/coop/rchain/rspace/nextgenrspace/history/RootsStore.scala

```diff
@@ -25,25 +25,39 @@ object RootsStoreInstances {
       ByteVector("current-root".getBytes(StandardCharsets.UTF_8)).toDirectByteBuffer
 
     override def currentRoot(): F[Option[Blake2b256Hash]] =
-      store
-        .get(currentRootName)
-        .map(_.map(bytes => Blake2b256Hash.codecBlake2b256Hash.decode(BitVector(bytes)).get.value))
+      for {
+        byteBuf ← store.get(currentRootName)
+        maybeDecoded ← byteBuf
+                        .map(
+                          bytes ⇒
+                            Blake2b256Hash.codecBlake2b256Hash
+                              .decode(BitVector(bytes))
+                              .get
+                        )
+                        .sequence
```

Comment:
I think that the `sequence` and `traverse` call for a helper

Reply:
thnx, will do.